snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java

```diff
@@ -83,6 +85,21 @@ public void process(StagingArea staging, RevisionSearcher searcher) throws IOExc
 				changedMembersByReferencedComponentId.put(((SnomedRefSetMemberIndexEntry) diff.newRevision).getReferencedComponentId(), diff);
 			});
 			
+			final ImmutableSet<String> valueIds = ImmutableSet.of(Concepts.CONCEPT_NON_CURRENT);
+			final Expression expression = Expressions.builder()
+					.filter(SnomedRefSetMemberIndexEntry.Expressions.referenceSetId(Concepts.REFSET_DESCRIPTION_INACTIVITY_INDICATOR))
```

Comment:
Can we restrict the number loaded members to just the necessary set of members?
How about restricting to the actually read values, descriptions in the transaction?